addons/mail/tests/test_mail_activity.py

```diff
@@ -18,3 +25,170 @@ def test_chatter_activity_tour(self):
             "mail_activity_schedule_from_chatter",
             login="admin",
         )
+
+
+@tests.tagged('mail_activity')
+class TestMailActivity(MailCommon):
+    @classmethod
+    def setUpClass(cls):
```

Comment:
There is a lot of setup, data and custom methods ... did not read in details (just first validation), but try to keep testing code as light and understandable as possible :) especially use clear input / output / expected values, as we end up not sure if we test the implementation of the testing methods that could have potential bugs.

For example the "todo_display_done / call_display_done" leads to a lot of tests that I am unsure is adding to the coverage. If we test a 'complete' use case with display_done activities, we are good to do. Maybe one small unit test without display_done just to be sure if does not crash in that case, but when possible it is better to have one test covering the most complicated use case than trying all possible combinations :)